CorsixTH/Lua/hospital.lua

```diff
@@ -1762,6 +1765,101 @@ function Hospital:countRoomOfType(type)
   return result
 end
 
+--! Get the number of reception desks in the hospital.
+--!return (int) Number of reception desks in the hospital.
+function Hospital:countReceptionDesks()
+  -- TODO Breaks in multiplayer mode.
+  return self.world.object_counts["reception_desk"]
+end
+
+--! Get the number of benches in the hospital.
+--!return (int) Number of benches in the hospital.
+function Hospital:countBenches()
+  -- TODO Breaks in multiplayer mode.
+  return self.world.object_counts["bench"]
+end
+
+--! Get the number of radiators in the hospital.
+--!return (int) Number of radiators in the hospital.
+function Hospital:countRadiators()
+  -- TODO Breaks in multiplayer mode.
+  return self.world.object_counts["radiator"]
+end
+
+--! Get the number of plants in the hospital.
+--!return (int) Number of plants in the hospital.
+function Hospital:countPlants()
+  -- TODO Breaks in multiplayer mode.
+  return self.world.object_counts["plant"]
+end
+
+--! Get the number of fire extinguishers in the hospital.
+--!return (int) Number of fire extinguishers in the hospital.
+function Hospital:countFireExtinguishers()
+  -- TODO Breaks in multiplayer mode.
+  return self.world.object_counts["extinguisher"]
+end
+
+--! Get the number of general objects in the hospital.
+--!return (int) Number of general objects in the hospital.
+function Hospital:countGeneralObjects()
+  -- TODO Breaks in multiplayer mode.
+  return self.world.object_counts["general"]
+end
+
+--! A new object has been placed in the hospital.
+--!param entity (Entity) The entity that was just placed.
+--!param id (string) That entity's id.
+function Hospital:objectPlaced(entity, id)
+  -- If it is a bench we're placing, notify queueing patients in the vicinity
+  if id == "bench" then
+    local notify_distance = 6
+    local w, h = self.world.map.th:size()
+    for tx = math.max(1, entity.tile_x - notify_distance), math.min(w, entity.tile_x + notify_distance) do
+      for ty = math.max(1, entity.tile_y - notify_distance), math.min(h, entity.tile_y + notify_distance) do
+        for _, patient in ipairs(self.world.entity_map:getHumanoidsAtCoordinate(tx, ty)) do
+          if class.is(patient, Patient) then
+            patient:notifyNewObject(id)
+          end
+        end
+      end
+    end
+  end
+
+  if id == "reception_desk" then
+    if self:isPlayerHospital() then
```

Comment:
It can do that player check once indeed. It can also return after the first match indeed.
Nothing here is however anywhere near critical, 3 tests won't kill performance, in particular as this code is only run when you place an object, which is almost never.